src/features/game/state/GameStateManager.ts

```diff
@@ -454,6 +458,55 @@ class GameStateManager {
     return this.checkpointTask.getAllVisibleTaskData();
   }
 
+  ///////////////////////////////
+  //          Quiz             //
+  ///////////////////////////////
+
+  /**
+   * Checks whether a quiz has been obtained full marks.
+   *
+   * @param key quiz id
+   * @returns {boolean}
+   */
+  public isQuizComplete(quizId: string): boolean {
+    return this.quizScores.get(quizId) === GameGlobalAPI.getInstance().getQuizLength(quizId);
+  }
+
+  /**
+   * Checks whether a specific quiz has been played.
+   *
+   * @param key quiz id
+   * @returns {boolean}
+   */
+  public isQuizAttempted(quizId: string): boolean {
+    return this.quizScores.has(quizId);
+  }
+
+  /**
+   * Get the score of a quiz.
+   * Return 0 if the quiz has not been played.
+   *
+   * @param quizId
+   * @returns
+   */
+  public getQuizScore(quizId: ItemId): number {
+    const score = this.quizScores.get(quizId);
+    if (score) {
+      return score;
+    }
+    return 0;
```

Comment:
Let's use optional chaining instead to simplify the code.


```suggestion
    return score ?? 0;
```